### PR TITLE
fix(wrapper/platform): assign product state to old prop

### PR DESCRIPTION
**Pull request by rxri**
fixes productstate on `1.2.37`

### CODE REVIEW

**Review by theRealPadster — APPROVED** [review]

File: jsHelper/spicetifyWrapper.js [hunk 1]
```diff
@@ -921,6 +934,23 @@ Spicetify.Events = (() => {
 		playerState.cache = playerState.current;
 	});
 
+	(function waitProductStateAPI() {
+		if (!Spicetify.Platform?.UserAPI) {
+			setTimeout(waitUserAPI, 100);
```
Comment (Delusoire) [inline jsHelper/spicetifyWrapper.js:939]: waitUserAPI is called regardless of this, so is it really needed here?

File: jsHelper/spicetifyWrapper.js [hunk 2]
```diff
@@ -921,6 +934,23 @@ Spicetify.Events = (() => {
...
+			return;
+		}
+
+		const productState = Spicetify.Platform.UserAPI?._product_state || Spicetify.Platform.UserAPI?._product_state_service;
```
Comment (Delusoire) [inline jsHelper/spicetifyWrapper.js:943]: The `if (!Spicetify.Platform?.UserAPI) { /* ... */; return }` check prevents `Spicetify.Platform.UserAPI` from ever being null or undefined, so the optional chaining is not needed.

Comment (rxri) [reply jsHelper/spicetifyWrapper.js:943]: Not valid. The prop can be undefined so it's better to stay like this

Comment (Delusoire) [reply jsHelper/spicetifyWrapper.js:943]: `a?.b` is equivalent to `a === null || a === undefined ? undefined : a.b`, b is irrelevant from the operator's perspective. The fact that `_product_state` and `_product_state_service` can be undefined changes nothing.

Comment (rxri) [reply jsHelper/spicetifyWrapper.js]: right... mb

**Review by Delusoire — APPROVED** [review]

**Pull request merged**